contrib/test.sh

```diff
@@ -2,6 +2,15 @@
 
 set -ex
 
+if [ "$DO_LINT" = true ]
+then
+    cargo clippy --all-features --all-targets -- -D warnings
+    cargo clippy --example bip32 -- -D warnings
+    cargo clippy --example handshake --features=rand-std -- -D warnings
+    cargo clippy --example ecdsa-psbt --features=bitcoinconsensus -- -D warnings
+    cargo clippy --example taproot-psbt --features=rand-std,bitcoinconsensus -- -D warnings
+fi
+
```

Comment:
We already have these lines in the `bitcoin/contrib/test.sh` script, what is the purpose of adding them here?

Reply:
This ones have `-D warnings` added.

Reply:
We have `-D warnings` in `bitcoin/contrib/test.sh` also